src/test/scala/DSATest.scala

```diff
@@ -0,0 +1,707 @@
+
+
+  }
+
+  test("interproc pointer arithmetic main") {
+    val results = RunUtils.loadAndTranslate(
+      BASILConfig(
+        loading = ILLoadingConfig(
+          inputFile = "examples/interproc_pointer_arithmetic/interproc_pointer_arithmetic.adt",
+          relfFile = "examples/interproc_pointer_arithmetic/interproc_pointer_arithmetic.relf",
+          specFile = None,
+          dumpIL = None,
+        ),
+        staticAnalysis = Some(StaticAnalysisConfig()),
+        boogieTranslation = BoogieGeneratorConfig(),
+        outputPrefix = "boogie_out",
+      )
+    )
+    val program = results.ir.program
+    val dsg = results.analysis.get.locals.get(program.mainProcedure)
+    val stack0 = dsg.adjust(dsg.find(dsg.stackMapping(0).cells(0)).getPointee)
+    val stack8 = dsg.adjust(dsg.find(dsg.stackMapping(8).cells(0)).getPointee)
+    val stack24 = dsg.adjust(dsg.find(dsg.stackMapping(24).cells(0)).getPointee)
+    val stack32 = dsg.adjust(dsg.find(dsg.stackMapping(32).cells(0)).getPointee)
+    val stack40 = dsg.adjust(dsg.find(dsg.stackMapping(40).cells(0)).getPointee)
+
+    assert(stack0.equals(dsg.adjust(dsg.formals(R29))))
+    assert(stack8.equals(dsg.adjust(dsg.formals(R30))))
+
+    // stack24 and 32 point to different offsets of the same node
+    assert(stack24.node.get.equals(stack32.node.get))
+    assert(stack24.offset == 0)
+    assert(stack32.offset == 16)
+
+    // stack40 points to a different offset of stack24's node but the analysis can't determine that in the local phase
+    assert(stack40._pointee.isDefined)
+    assert(!stack40.node.get.equals(stack24.node.get))
+  }
+
+  test("interproc pointer arithmetic callee") {
+    val results = RunUtils.loadAndTranslate(
+      BASILConfig(
+        loading = ILLoadingConfig(
+          inputFile = "examples/interproc_pointer_arithmetic/interproc_pointer_arithmetic.adt",
+          relfFile = "examples/interproc_pointer_arithmetic/interproc_pointer_arithmetic.relf",
+          specFile = None,
+          dumpIL = None,
+        ),
+        staticAnalysis = Some(StaticAnalysisConfig()),
+        boogieTranslation = BoogieGeneratorConfig(),
+        outputPrefix = "boogie_out",
+      )
+    )
+    val program = results.ir.program
+    val dsg = results.analysis.get.locals.get(program.procs("callee"))
+    val stack8 = dsg.adjust(dsg.find(dsg.stackMapping(8).cells(0)).getPointee)
+    val stack24 = dsg.adjust(dsg.find(dsg.stackMapping(24).cells(0)).getPointee)
+
+    // stack8 points to the formal argument object
+    assert(stack8.equals(dsg.adjust(dsg.formals(R0))))
+    assert(stack8.offset == 0)
+    // stack 24 points to the formal argument object at offset 16, instead
+    assert(stack24.equals(dsg.adjust(dsg.formals(R0)).node.get.cells(16)))
+  }
+
+
+  test("internal merge") {
+    // this is an internal merge (two cells of the same node overlap and are merged together)
+    val mem = SharedMemory("mem", 10000, 10000)
```

Comment:
These are not correct parameters - the addressSize should be 64 (since addresses are 64-bit) and the valueSize should be 8 (since 8 bits are stored at every offset).